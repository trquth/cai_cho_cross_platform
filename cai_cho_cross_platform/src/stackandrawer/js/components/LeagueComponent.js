import React, { PropTypes, Component } from 'react';
import { connect } from 'react-redux';
import { View, Text, FlatList, TouchableOpacity, StatusBar } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import Header from '../components/Header';
import { getTeam } from '../actions/team';

class LeagueComponent extends Component {

  static propTypes = {
    league: PropTypes.shape().isRequired,
    getTeam: PropTypes.func.isRequired,
  }

  renderTeam = ({ item, index }) => (
    <TouchableOpacity
      testID={`Team.${item.name}`}
      style={{ backgroundColor: index % 2 ? 'white' : '#F6F6F6' }}
      onPress={() => this.props.getTeam(this.props.league.routeName, item.name)}
    >
      <View style={styles.itemContainer}>
        <Text>{item.name}</Text>
        <Icon name={'ios-arrow-forward'} color={'black'} size={20} />
      </View>
    </TouchableOpacity>
  );

  render() {
    const { league } = this.props;
    const { renderTeam } = this;

    return (
      <View style={styles.container}>
        <StatusBar barStyle={'light-content'} />
        <Header title={league.name} image={null} />
        <FlatList
          data={league.clubs}
          renderItem={renderTeam}
        />
      </View>
    );
  }
}

const styles = {
  itemContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 16,
    paddingVertical: 10,
    flex: 1,
  },
  container: {
    flex: 1,
  },
};

const mapStateToProps = ({ league }) => ({ league: league.teams });
export default connect(mapStateToProps, { getTeam })(LeagueComponent);
