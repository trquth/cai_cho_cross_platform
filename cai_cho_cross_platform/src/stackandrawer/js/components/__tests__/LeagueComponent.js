import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import mockStore from 'redux-mock-store';
import { FlatList } from 'react-native';
import { expect as expectChai } from 'chai';
import LeagueComponent from '../LeagueComponent';

// eslint-disable-next-line
const league = {"name":"English Premier League 2016/17","routeName":"PremierLeague","clubs":[{"key":"chelsea","name":"Chelsea","code":"CHE"},{"key":"arsenal","name":"Arsenal","code":"ARS"},{"key":"tottenham","name":"Tottenham Hotspur","code":"TOT"},{"key":"westham","name":"West Ham United","code":"WHU"},{"key":"crystalpalace","name":"Crystal Palace","code":"CRY"},{"key":"manutd","name":"Manchester United","code":"MUN"},{"key":"mancity","name":"Manchester City","code":"MCI"},{"key":"everton","name":"Everton","code":"EVE"},{"key":"liverpool","name":"Liverpool","code":"LIV"},{"key":"westbrom","name":"West Bromwich Albion","code":"WBA"},{"key":"stoke","name":"Stoke City","code":"STK"},{"key":"sunderland","name":"Sunderland","code":"SUN"},{"key":"southampton","name":"Southampton","code":"SOU"}]};

it('renders all children', () => {
  const initialState = { league: { teams: league } };
  const store = mockStore(initialState);

  const wrapper = shallow(<LeagueComponent store={store} />);

  const component = shallow(wrapper.getNodes()[0]);

  expectChai(component.find(FlatList).length).to.equal(1);
  expectChai(component.find(FlatList).at(0).props().data.length).to.equal(league.clubs.length);
  expect(toJson(component)).toMatchSnapshot();
});
