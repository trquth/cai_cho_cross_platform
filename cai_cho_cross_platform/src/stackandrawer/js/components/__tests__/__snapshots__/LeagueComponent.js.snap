// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all children 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    barStyle="light-content"
    showHideTransition="fade"
  />
  <Connect(Header)
    image={null}
    title="English Premier League 2016/17"
  />
  <FlatList
    data={
      Array [
        Object {
          "code": "CHE",
          "key": "chelsea",
          "name": "Chelsea",
        },
        Object {
          "code": "ARS",
          "key": "arsenal",
          "name": "Arsenal",
        },
        Object {
          "code": "TOT",
          "key": "tottenham",
          "name": "Tottenham Hotspur",
        },
        Object {
          "code": "WHU",
          "key": "westham",
          "name": "West Ham United",
        },
        Object {
          "code": "CRY",
          "key": "crystalpalace",
          "name": "Crystal Palace",
        },
        Object {
          "code": "MUN",
          "key": "manutd",
          "name": "Manchester United",
        },
        Object {
          "code": "MCI",
          "key": "mancity",
          "name": "Manchester City",
        },
        Object {
          "code": "EVE",
          "key": "everton",
          "name": "Everton",
        },
        Object {
          "code": "LIV",
          "key": "liverpool",
          "name": "Liverpool",
        },
        Object {
          "code": "WBA",
          "key": "westbrom",
          "name": "West Bromwich Albion",
        },
        Object {
          "code": "STK",
          "key": "stoke",
          "name": "Stoke City",
        },
        Object {
          "code": "SUN",
          "key": "sunderland",
          "name": "Sunderland",
        },
        Object {
          "code": "SOU",
          "key": "southampton",
          "name": "Southampton",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    removeClippedSubviews={true}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
